src/odemis/util/driver.py

```diff
@@ -209,15 +209,15 @@ def guessActuatorMoveDuration(actuator, axis, distance, accel=DEFAULT_ACCELERATI
     distance (float): the move for which the time to be estimated  
     return (float >= 0): the estimated time (in s)
     """
-    speed = None
     if not (hasattr(actuator, "axes") and isinstance(actuator.axes, dict)):
         raise ValueError("The component %s should be an actuator, but it is not." % actuator)
-    if not axis in actuator.axes:
+    if axis not in actuator.axes:
         raise KeyError("The actuator component %s is expected to have %s axis, but it does not." % (actuator, axis))
+
+    speed = DEFAULT_SPEED
     if model.hasVA(actuator, "speed"):
-        speed = actuator.speed.value.get("z", None)
-    if speed is None:
-        speed = DEFAULT_SPEED
+        speed = actuator.speed.value.get(axis, DEFAULT_SPEED)
```

Comment:
@pieleric I've updated this function, let me know if you agree, then I will use this function

Reply:
Yes. Looks fine. Actually, Stefan had already fixed the "z" hard-coded... but it's not merged yet, that's why you didn't see it.